src/brogue/Monsters.c

```diff
@@ -334,6 +334,20 @@ boolean monstersAreTeammates(const creature *monst1, const creature *monst2) {
                  && monst1->leader == monst2->leader)) ? true : false);
 }
 
+/// @brief Checks if a monster always reflects bolts (guardian spirit, mirror totem, stone/winged guardian)
+/// These monsters also always reflect the bolt back at the caster.
+/// @param monst the monster
+/// @return true if the monster always reflect bolts
+boolean monsterAlwaysReflectsBolts(const creature *monst) {
+    if (monst
+        && (monst->info.flags & MONST_REFLECT_4) 
```

Comment:
Surprisingly no. It's only the combination of MONST_REFLECT_4 and MONST_ALWAYS_USE_ABILITY that always reflects. MONST_REFLECT_4 essentially means the monster wears +4 armor of reflection.

I thought about creating a new flag MONST_ALWAYS_REFLECTS instead, but didn't because I felt like it was scope creep. I do think it's worth doing though. I'll create a new issue for it if you agree.